metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -718,23 +714,6 @@ def _init_tableau_project_registry(self, all_project_map: dict) -> None:
             logger.debug(f"Project {project.name} is added in project registry")
             self.tableau_project_registry[project.id] = project
 
-        if self.config.extract_project_hierarchy is False:
-            logger.debug(
-                "Skipping project hierarchy processing as configuration extract_project_hierarchy is "
-                "disabled"
-            )
-            return
-
-        logger.debug("Reevaluating projects as extract_project_hierarchy is enabled")
-
-        for project in list_of_skip_projects:
-            if (
-                project.parent_id in self.tableau_project_registry
```

Comment:
Maybe I'm missing something but from my understanding, this can never happen.
If we only add projects to `list_of_skip_projects` that are not allowed there will never be a project in this list that is _not denied_ as it is checked below.

When checking if a project is allowed, the following is executed downstream https://github.com/datahub-project/datahub/blob/ebe7b2dc5a0fa0b72e926ce291052cbdbb08f75b/metadata-ingestion/src/datahub/configuration/common.py#L261-L262 and this also checks it's denied.

What do you think? Am I missing something here?

Reply:
It might have been implemented based on user requests to include child projects by default unless explicitly denied. Could you please revert these changes?

Reply:
That's the thing, I think we always include child projects with or without the property `extract_project_hierarchy`.  So I don't really understand why we need it. From what I understand, the only use case, where this property actually has an impact is the one I mentioned above:

> So if for a project the path or the name is not allowed but not explicitly denied, this part here would still add it if the parent project was added (and extract_project_hierarchy was enabled).

I can revert the changes, but IMO we should have a further look at this part as it does things that are hard to understand, at least for me. Could be that as the code evolved over time, we don't even need it anymore.
Again, maybe I'm missing something and for some reason this makes total sense, but it could be worth checking again. @sid-acryl could you do that together with @hsheth2?